### PR TITLE
add helpers for search fields

### LINKED ISSUE #10833: Add helpers for Search Index Fields
Roughly along these lines:

https://gist.github.com/bryevdv/07246cb8f1186548bc1a182b7ff4b93f

### PR DESCRIPTION
**Pull request by bryevdv**
close #10833

### CODE REVIEW

File: sdk/search/azure-search-documents/azure/search/documents/_service/_index.py [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+if TYPE_CHECKING:
+    from typing import Any, Dict, List
+
+__all__ = ("ComplexField", "SearchableField", "SimpleField")
+
+
+def SimpleField(
+    name,
+    type,  # pylint:disable=redefined-builtin
+    key=False,
```
Comment (rakshith91) [inline sdk/search/azure-search-documents/azure/search/documents/_service/_index.py:20]: Obviously we can change these later - but would like most of these being keyword only

Comment (johanste) [reply sdk/search/azure-search-documents/azure/search/documents/_service/_index.py:20]: They should be keyword only arguments, yes. There is no logical ordering between them as far as I can see.

File: sdk/search/azure-search-documents/azure/search/documents/_service/_index.py [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+    filterable=False,
+    facetable=False,
+    sortable=False,
+    hidden=False,
+):
+    # type: (str, str, bool, bool, bool, bool, bool, **Any) -> Dict[str, Any]
```
Comment (johanste) [inline sdk/search/azure-search-documents/azure/search/documents/_service/_index.py:26]: `**Any`?

File: sdk/search/azure-search-documents/azure/search/documents/_service/_index.py (hunk 1 above)
Comment (bryevdv) [reply sdk/search/azure-search-documents/azure/search/documents/_service/_index.py:20]: https://github.com/Azure/azure-sdk-for-python/issues/11188

File: sdk/search/azure-search-documents/azure/search/documents/_service/_index.py (hunk 2 above)
Comment (bryevdv) [reply sdk/search/azure-search-documents/azure/search/documents/_service/_index.py:26]: Why is this passing mypy?

Comment (rakshith91) [reply sdk/search/azure-search-documents/azure/search/documents/_service/_index.py]: seems like we did not hard enable mypy check - it's going to pass no matter what.

**Pull request merged**